libs/grout/src/util.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable vx/gts-jsdoc */
+
+export function isBoolean(value: unknown): value is boolean {
+  return typeof value === 'boolean';
+}
+
+export function isNumber(value: unknown): value is number {
+  return typeof value === 'number';
+}
+
+export function isString(value: unknown): value is string {
+  return typeof value === 'string';
+}
+
+export function isArray(value: unknown): value is unknown[] {
+  return Array.isArray(value);
+}
+
+export function isObject(value: unknown): value is Record<string, unknown> {
+  return typeof value === 'object' && value !== null;
+}
```

Comment:
Worth noting that this function will return true when passed an array, which might be misleading given the typeguard

Reply:
ea098eef0